ui/src/app/base/validation.test.ts

```diff
@@ -0,0 +1,21 @@
+import { hostnamePattern } from "./validation";
+
+describe("hostname validation", () => {
```

Comment:
Could you add length tests for coverage and check the error label is returned as expected?

Reply:
Done.